core/lib/zksync_core/src/sync_layer/external_io.rs

```diff
@@ -220,34 +217,31 @@ impl StateKeeperIO for ExternalIO {
         max_wait: Duration,
     ) -> anyhow::Result<Option<L1BatchParams>> {
         tracing::debug!("Waiting for the new batch params");
-        for _ in 0..poll_iters(POLL_INTERVAL, max_wait) {
-            match self.actions.pop_action() {
-                Some(SyncAction::OpenBatch {
-                    params,
-                    number,
-                    first_miniblock_number,
-                }) => {
-                    anyhow::ensure!(
-                        number == cursor.l1_batch,
-                        "Batch number mismatch: expected {}, got {number}",
-                        cursor.l1_batch
-                    );
-                    anyhow::ensure!(
-                        first_miniblock_number == cursor.next_miniblock,
-                        "Miniblock number mismatch: expected {}, got {first_miniblock_number}",
-                        cursor.next_miniblock
-                    );
-                    return Ok(Some(params));
-                }
-                Some(other) => {
-                    anyhow::bail!("unexpected action in the action queue: {other:?}");
-                }
-                None => {
-                    tokio::time::sleep(POLL_INTERVAL).await;
-                }
+        let Some(action) = self.actions.recv_action(max_wait).await else {
+            return Ok(None);
+        };
+        match action {
+            SyncAction::OpenBatch {
```

Comment:
Bikeshedding: I'd retain matching on the `Option`al action instead of a let-else above, IMO, it's a tiny bit more readable.